File: src/eu/yeger/gramofo/fol/FOLToken.java
package eu.yeger.gramofo.fol;

public class FOLToken {
	
	public static final int END_OF_SOURCE = 0;
	public static final int SYMBOL = 1;
	public static final int CHAR = 2;
	public static final int TRUE =  3;
	public static final int FALSE =  4;
	public static final int OR =  5;
	public static final int AND =  6;
	public static final int NOT =  7;
	public static final int IMPLICATION =  8;
	public static final int BIIMPLICATION =  9;
	public static final int EXISTS =  10; 
	public static final int FORALL =  11;
	public static final int INFIX_PRED =  12;
	public static final int EQUAL_SIGN =  13;
	public static final int INFIX_FUNC =  14;
	public static final int BRACKET = 15;
	public static final int DOT = 16;
	public static final int COMMA = 17;

	
	private int type;
	private String value;
	
	public FOLToken(int type, String value) {
		this.type = type;
		this.value = value;
	}
	
	public int getType() {
		return type;
	}
	
	public String getValue() {
		return value;
	}

	public void setTypeAndValue(int type, String value) {
		this.type = type;
		this.value = value;
	}
}

File: src/eu/yeger/gramofo/fol/parser/FOLToken.kt
package eu.yeger.gramofo.fol.parser

class FOLToken(var type: Int, var value: String) {
    fun setTypeAndValue(type: Int, value: String) {
        this.type = type
        this.value = value
    }

    companion object {
        const val END_OF_SOURCE = 0
        const val SYMBOL = 1
        const val CHAR = 2
        const val TRUE = 3
        const val FALSE = 4
        const val OR = 5
        const val AND = 6
        const val NOT = 7
        const val IMPLICATION = 8
        const val BI_IMPLICATION = 9
        const val EXISTS = 10
        const val FOR_ALL = 11
        const val INFIX_PRED = 12
        const val EQUAL_SIGN = 13
        const val INFIX_FUNC = 14
        const val BRACKET = 15
        const val DOT = 16
        const val COMMA = 17
    }
}
